code-challenges.test.js

```diff
@@ -30,33 +51,93 @@ var secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+//declare a functions named codedMessage taking in a string
+//initiate variable of a newString
+//write a for loop going through each string
+//write conditional statement if numbers equals vowels then 
+//access the values "a, e, i, o, u" and change them to numbers.
+////use the .split() to split the string
 
+const codedMessage = (string) => {
+    let newString = []
+    for(i=0; i<string.length; i++){
+        if(string[i] == "a", "e", "i", "o", "u"){
+            newString.split([i])
+        }
+        return newString
+    }
+}
+console.log(codedMessage(secretCodeWord1)); //Test didn't pass. Ended up getting stuck and moved on
 
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
 // a) Create a test with an expect statement using the variable provided.
 
+//describe a function named returnA and takes in an array
+//use let as a place holder for the variables provided
+//write the result using it
+//write expect with the functions and variables toEqual the espected outputs 
+
+
+describe("returnA", () => {
+    let arrayOfWords1 = ["Apple", "Banana", "Plum", "Cherry", "Kiwi"]
+    let arrayOfWords2 = ["Mango", "Orange", "Apricot", "Blueberry", "Peach"]
+    it("returns all the words that contain the letter a",() => {
+        expect(returnA(arrayOfWords1)).toEqual(["Apple", "Banana"])
+        expect(returnA(arrayOfWords2)).toEqual(["Mango", "Orange", "Apricot", "Peach"])
+    })
+})
+
+
 var arrayOfWords1 = ["Apple", "Banana", "Plum", "Cherry", "Kiwi"]
 // Expected output: ["Apple", "Banana"]
 var arrayOfWords2 = ["Mango", "Orange", "Apricot", "Blueberry", "Peach"]
-// Expected output: ["Mango", "Banana", "Peach"]
-
+// Expected output: ["Mango", "Orange", "Apricot", "Peach"]
 
 
 
 
 // b) Create the function that makes the test pass.
 
+//declare a function anems returnA that returns an array
+//initiate a variable for the new array
+//loops through the original array
+//for the split method, I have having trouble figuring out how to implement it in the code. I understand I need to split the array and every letter in the words then return the letters that are not being taken out.
 
+const returnA = (array) => {
+    let newArray = []
+    for(i=0; i<array.length; i++){
+        newArray.split(arrayOfWords1)
+    }
+    return newArray
```

Comment:
Here you are returning your empty new array. the for loop doesn't actually do anything here because its running split ( a string method) on your local variable newArray (an empty array). 

I can see you have a tool you like to use, namely the for loop. I don't think it's the right tool for the job here.  I think that's relying heavily on the for loop is leading you down the wrong path.